modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -61,7 +73,12 @@ def is_module_platform?(mod)
   end
 
   def is_session_type_compat?(mod)
-    mod.session_compatible? session.sid
+    validation_opts = {
+      validate_arch: datastore['ValidateArch'],
+      validate_platform: datastore['ValidatePlatform'],
+      validate_meterpreter_commands: datastore['ValidateMeterpreterCommands']
+    }
+    mod.session_compatible?(session.sid, validation_opts)
```

Comment:
Could this avoid introducing new options on the `session_compatible?` method?  Consider inspecting the reasons returned:
```suggestion
    reasons = mod.session_compatible?(session.sid)
    valid_reasons = []
    reasons.each do |reason|
       if datastore['ValidateArch'] && reason.include?('architecture')
          print_status "#{mod} included with override of #{reason}"
          next
       end
       if datastore['ValidatePlatform'] && reason.include?('platform')
          print_status "#{mod} included with override of #{reason}"
          next
       end
       if datastore['ValidateMeterpreterCommands'] && (reason.start_with?('missing') || reason.start_with?('unloadable'))
          print_status "#{mod} included with override of #{reason}"
          next
       end
       valid_reasons << reason
    end
    return valid_reasons.empty?
```

The benefit here being that this would print out module that are of dubious compatibility.  I am not enthused about recommending string inspection here as it could end up brittle, however the trade off in visibility for the user may be helpful.

Reply:
We discussed that a bit, but didn't want to do the string comparisons either - and didn't want to block this PR on rewriting the API here to return various enum values etc.

I think we should be be good to ship this as-is to unbreak the local exploit suggester before tomorrow's release, and we can circle back to improving the debug capabilities of this module as a second pass as well. We were thinking a simple rex table hidden behind a datastore option might also aid in debugging things when we know a module should be suggested, but isn't. We should try to find the right balance of information here of course.

Reply:
One concern here is that the `exceptions` are being lost in the `post_mixin` as is.  Unless we call `session_compatible?` twice here with and without the filters we cannot see that a module was included only because the validation was suppressed.

Reply:
IMO iteration should start with least impact to the core codebase.  Adding the injection of options to `session_compatible?` instead of keeping changes in the exploit suggester module may cause wider ranging churn.  Least impact seems especially important if the goal is to get something working then improve.

Reply:
Will snip out just the metadata removal to a fresh PR https://github.com/rapid7/metasploit-framework/pull/16455/files, and we can iterate on this PR

Reply:
I'll just chime in here as a tie breaker of sorts and say that I agree with Jeffery's points on this and I think these are valid things that we should address here. Whether or not this means we should delay the landing of this PR though is another matter and unfortunately one that I think I should have the ultimate say on.